### PR TITLE
Task/gcndsp 306 add get transparency site to api

### PR DESCRIPTION
**Pull request by leew**
Implemented the new API functionality

### CODE REVIEW

File: modules/api/src/Service/TransparencySiteService.cs [hunk 1]
```diff
@@ -66,5 +67,25 @@ private DynamicParameters BuildBlankSearchParameters()
...
+
+    public Task<TransparencySite> GetSiteAsync(Guid id)
+    {
+      var query = "application.find_site";
+
+      var parameters = new DynamicParameters();
+      parameters.Add("_site_unique_identifier", id, DbType.Guid, ParameterDirection.Input);
+
+      return _dataService.ExecuteQueryFirstOrDefault<TransparencySite>(query, parameters);
+      // if (id == "test")
```
Comment (ChristopherJamesMorris) [inline modules/api/src/Service/TransparencySiteService.cs]: Can we remove any commented out code?

File: modules/api/src/Validators/ITransparencySiteRequestValidator.cs [hunk 2]
```diff
@@ -4,5 +4,7 @@ namespace GpConnect.NationalDataSharingPortal.Api.Validators;
 
 public interface ITransparencySiteRequestValidator
```
Comment (ChristopherJamesMorris) [inline modules/api/src/Validators/ITransparencySiteRequestValidator.cs]: Can we stick all the interfaces in a separate folder to keep them separate from the implementations?

File: database/functions/R__application.find_site.sql [hunk 3]
```diff
@@ -0,0 +1,79 @@
+--
+-- Name: find_site(uuid); Type: FUNCTION; Schema: application; Owner: postgres
+--
+
+CREATE FUNCTION application.find_site(
```
Comment (ChristopherJamesMorris) [inline database/functions/R__application.find_site.sql:5]: We're going to need to do some refactoring of this and the find_sites DB function but we can do this under a separate ticket in the next sprint or something.

Comment (leew) [reply database/functions/R__application.find_site.sql:5]: Ok, sounds good to me.

Comment (leew) [reply database/functions/R__application.find_site.sql:5]: Makes sense to me

File: modules/api/src/Service/TransparencySiteService.cs (hunk 1 above)
Comment (leew) [reply modules/api/src/Service/TransparencySiteService.cs]: Good spot, Thanks @ChristopherJamesMorris

File: modules/api/src/Validators/ITransparencySiteRequestValidator.cs (hunk 2 above)
Comment (leew) [reply modules/api/src/Validators/ITransparencySiteRequestValidator.cs]: Interaces migrated as part of fa51e77b6f905ef50ab35d3a0ce9d1f10cc40f8e, all tests continue to pass

**Review by ChristopherJamesMorris — APPROVED** [review]
All good - thanks.

**Pull request merged**